### PR TITLE
Add explanation how to setup development environment and fix one test

### PR DESCRIPTION
**Pull request by adrianbrink**
I've added a small paragraph about setting up your development environment. It explains things like that you need solidity and that you need to install the requirements-dev.txt. 

Secondly, this PR fixes an import statement in test_encoding.py. 

Lastly, I am curious whether there should be that many tests failing or whether I have setup my environment incorrectly :-)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -17,6 +17,12 @@ and what are the requirements for a Pull Request to be opened against Raiden.
 There are two ways you can contribute to the development. You can either open
 an Issue or if you have programming abilities open a Pull Request.
 
+## Setup development environment
+- Run ```pip install -r requirements-dev.txt -e .``` (it is highly recommended that you use something like pyenv and pyenv-virtualenv to manage the python dependencies)
+- Install [solidity](https://solidity.readthedocs.io/en/latest/installing-solidity.html) and ensure that ```solc --version``` prints out something
+- Run ```make test``` before making any changes to ensure that everything works so far
+- Currently I bunch of tests are failing - *2 failed, 108 passed, 1 skipped, 6 xfailed, 37 error*
```
Comment (LefterisJP) [inline CONTRIBUTING.md]: This is in your local environment.

The recommended way to run the tests would be with `tox`.

File: raiden/api/test_encoding.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 from raiden.channel import Channel, ChannelEndState, ChannelExternalState
 from raiden.utils import make_address, make_privkey_address
 
-from raiden.api.encoding import ChannelListSchema, ChannelSchema
```
Comment (LefterisJP) [inline raiden/api/test_encoding.py]: no need for this change. But thanks for actually notifying us of this test file. It should probably be deleted, or at the very least fixed, and put along with all other tests under the `tests/` directory.

File: CONTRIBUTING.md (hunk 1 above)
Comment (adrianbrink) [inline CONTRIBUTING.md]: Ah okay.

This is the [tox](https://tox.readthedocs.io/en/latest/), correct?

Should I update the CONTRIBUTING.md though with more information for new comers to get started quickly?

File: raiden/api/test_encoding.py (hunk 2 above)
Comment (adrianbrink) [inline raiden/api/test_encoding.py]: Oh right :-)

I'll delete it in an update to this PR.

**Comment by adrianbrink** [comment]
I am getting this error on a couple of tests: ```E           error: [Errno 49] Can't assign requested address: ('127.0.0.10', 29872)```. 

Do I need to do something else besides just running ```tox```?

**Comment by LefterisJP** [comment]
@adrianbrink Hmm the error does not ring a bell atm, will take a look. Also if anyone else from the team sees this and has an idea please comment.

One other thing about our processes. When you make a PR like this and want to syncrhonize it with master, we strongly encourage rebasing on master instead of merging master into the branch.

**Comment by LefterisJP** [comment]
@adrianbrink Also please don't touch the test_encoding.py in this PR. We also prefer small PRs with a distinct purpose, not PRs doing many different things.

**Comment by adrianbrink** [comment]
@LefterisJP 
Alright, that is good to know :-)

I have reverted the deletion of the file and also the merging of master and instead rebased.

However, it might have gone wrong, so please double check :-) even though to me it looks correct with ```git log```.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -17,6 +17,20 @@ and what are the requirements for a Pull Request to be opened against Raiden.
 There are two ways you can contribute to the development. You can either open
 an Issue or if you have programming abilities open a Pull Request.
 
+## Setup development environment
+### Installation
+- Install [pyenv](https://github.com/pyenv/pyenv) and [pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv) and create a separate virtual environment for your raiden installation
+- Make sure that [tox](https://tox.readthedocs.io/en/latest/) is installed
```
Comment (LefterisJP) [inline CONTRIBUTING.md]: tox is part of the development dependencies. No need to install manually.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -17,6 +17,20 @@ and what are the requirements for a Pull Request to be opened against Raiden.
...
+- Install [solidity](https://solidity.readthedocs.io/en/latest/installing-solidity.html) and ensure that ```solc --version``` prints out something
+
+### Dependencies
+- Run ```pip install -e .``` to install the dependencies
```
Comment (LefterisJP) [inline CONTRIBUTING.md]: To run the tests you also need the development requirements. These are  in `dev_requirements.txt`.

So the only one thing you need to do is `pip install -r dev_requirements.txt`

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -17,6 +17,20 @@ and what are the requirements for a Pull Request to be opened against Raiden.
...
+
+### Tests
+- Run ```tox``` to run the tests.
+// TODO: Fix that after this some tests are still failing.
```
Comment (LefterisJP) [inline CONTRIBUTING.md]: I'd rather not have an empty section with a TODO here. Tests may always fail locally, or be broken due to some mistake. That should not be reflected in the `README` file.

We have a strict rule of not accepting PRs if tests fail in the CI, so if tests fail locally it's probably something on your side.

File: raiden/api/test_encoding.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
+import random
```
Comment (LefterisJP) [inline raiden/api/test_encoding.py]: You are readding the `test_encoding.py` file here. Please remove.

File: raiden/smart_contracts/NettingChannelContract.sol [hunk 7]
```diff
@@ -114,37 +111,31 @@ contract NettingChannelContract {
...
     }
 
     /// @notice Returns the block number for when the channel was opened.
-    /// @return The block number for when the channel was opened.
     function opened() constant returns (uint) {
         return data.opened;
     }
 
     /// @notice Returns the block number for when the channel was closed.
-    /// @return The block number for when the channel was closed.
```
Comment (LefterisJP) [inline raiden/smart_contracts/NettingChannelContract.sol]: You are also removing comments from the code. This leads me to believe you rebased against an earlier version of master. Please rebase against the latest master.

**Comment by LefterisJP** [comment]
@adrianbrink Can you address my comments and fix the rebase while at the same time taking care of what I asked?

**Comment by hrishikeshio** [comment]
@adrianbrink I had the same issue. I solved it by following this small comment :) Should be in docs somewhere
https://github.com/raiden-network/raiden/blob/master/raiden/tests/utils/network.py#L258 

>  I am getting this error on a couple of tests: E error: [Errno 49] Can't assign requested address: ('127.0.0.10', 29872).
> 
> Do I need to do something else besides just running tox?

**Comment by LefterisJP** [comment]
@adrianbrink ping?

**Comment by hackaugusto** [comment]
closed because of inactivity

**Pull request closed without merging**